test/org/opensolaris/opengrok/authorization/AuthorizationFrameworkTest.java

```diff
@@ -22,326 +22,655 @@
  */
 package org.opensolaris.opengrok.authorization;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
 import javax.servlet.http.HttpServletRequest;
 import org.junit.AfterClass;
 import org.junit.Assert;
 import org.junit.BeforeClass;
 import org.junit.Test;
-import org.opensolaris.opengrok.authorization.AuthorizationCheck.AuthControlFlag;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 import org.opensolaris.opengrok.configuration.Group;
+import org.opensolaris.opengrok.configuration.Nameable;
 import org.opensolaris.opengrok.configuration.Project;
 import org.opensolaris.opengrok.configuration.RuntimeEnvironment;
 import org.opensolaris.opengrok.web.DummyHttpServletRequest;
 
-import static org.junit.Assert.assertEquals;
-
+@RunWith(Parameterized.class)
 public class AuthorizationFrameworkTest {
 
     private static String pluginDirectory;
-
```

Comment:
This file is heavily refactored and changed to easily form the test cases for the authorization and to support a test cases for substacks. All important test functions in the previous version have been converted to the array of parameters in the new version.